File: knowit/2022/day14/main.cpp
#include <iostream>

#define LENGTH 100000
#define LENGTH_PER_REINDEER 1000

int main() {
    int shifts = LENGTH/LENGTH_PER_REINDEER;
    unsigned long long weight = 900+100;
    unsigned long long reindeers = 0;
    for (int i = 0; i < shifts; i++) {
        weight += LENGTH_PER_REINDEER;
        unsigned long long new_reindeers = (weight+199)/200 + (reindeers+1)/2;
        reindeers += new_reindeers;
        /*for (int j = 0; j*200 < weight; j++) {
            new_weight += 100;
            reindeers++;
        }*/
        printf("%lld %lld %lld\n", weight, new_reindeers, reindeers);
        
    }
    printf("%lld\n", reindeers);
}

File: knowit/2022/day14/main.py
length = 100000
length_per_reindeer = 1000
shifts = length//length_per_reindeer

weight = 900+100
reindeers = 0
for i in range(shifts):
    weight += length_per_reindeer
    new_reindeers = (weight+199)//200 + (reindeers+1)//2
    reindeers += new_reindeers
    print(reindeers)